models/usage.go

```diff
@@ -0,0 +1,244 @@
+		return 0.16
+	default:
+		// this is a catch-all for free tier
+		// free tier users will never encounter a charge call
+		return 9999
+	}
+}
+
+var (
+	// Free is what every signed up user is automatically registered as
+	// Restrictions of free:
+	//			* No on-demand data encryption
+	//			* 3GB/month max
+	//			* IPNS limit of 5, with no automatic republishes
+	//			* 5 keys
+	Free DataUsageTier = "free"
+
+	// Partner is for partners of RTrade
+	// partners have 100GB/month free
+	//			* on-demand data encryption
+	//			* 0.16GB/month after 100GB limit
+	Partner DataUsageTier = "partner"
+
+	// Light is the first non-free, and non-partner tier
+	// tier is from 3GB -> 100GB
+	// after reaching 100GB they are upgraded to Plus
+	//			* on-demand data encryption
+	//			* 0.22
+	Light DataUsageTier = "light"
+
+	// Plus is the other non-free, non-partner, non-light tier
+	// tier is from 100GB -> 1TB to our max monthly usage of 1TB
+	// 			* on-demand data encryption
+	//			* $0.165
+	Plus DataUsageTier = "plus"
+)
+
+// Usage is used to handle Usage of Temporal accounts
+type Usage struct {
+	UserName string `gorm:"type:varchar(255);unique"`
+	// keeps track of the max monthly upload limit for the user
+	MonthlyDataLimitGB float64 `gorm:"type:float;default:0"`
+	// keeps track of the current monthyl upload limit used
+	CurrentDataUsedGB float64 `gorm:"type:float;default:0"`
+	// keeps track of how many IPNS records the user has published
+	IPNSRecordsPublished int64 `gorm:"type:integer;default:0"`
+	// keeps track of how many messages the user has sent
+	PubSubMessagesSent int64 `gorm:"type:integer;default:0"`
+	// Used to indicate whether or not the user
+	// has consumed their free private network trial
+	PrivateNetworkTrialUsed bool `gorm:"type:boolean;default:false"`
+	// Used to determine when their private network trial ends
+	// good until 2038 <- ticket is open in JIRA so we do not forget about this
+	// trial is 800 hours in unix nano
+	TrialEndTime int64 `gorm:"type:integer;default:0"`
+	// keeps track of the tier the user belongs to
+	Tier DataUsageTier `gorm:"type:varchar(255)"`
+}
+
+// UsageManager is used to manage Usage models
+type UsageManager struct {
+	DB *gorm.DB
+}
+
+// NewUsageManager is used to instantiate a Usage manager
+func NewUsageManager(db *gorm.DB) *UsageManager {
+	return &UsageManager{DB: db}
+}
+
+// NewUsageEntry is used to create a new usage entry in our database
+// if tier is free, limit to 3GB monthly otherwise set to 1TB
+func (bm *UsageManager) NewUsageEntry(username string, tier DataUsageTier) (*Usage, error) {
+	usage := &Usage{
+		UserName:                username,
+		CurrentDataUsedGB:       0,
+		IPNSRecordsPublished:    0,
+		PubSubMessagesSent:      0,
+		PrivateNetworkTrialUsed: false,
+		TrialEndTime:            0,
+		Tier:                    tier,
+	}
+	if tier == Free {
+		usage.MonthlyDataLimitGB = datasize.GB.GBytes()
+	} else {
+		usage.MonthlyDataLimitGB = datasize.TB.GBytes()
```

Comment:
what's going on here? (one using `GB`, another using `TB`)
also the documentation implies free gets 3gb but it seems to be just 1gb here

Reply:
So `datasize.GB.GBytes()` is 1GB represented as GB (you can also do bytes, etc..) while `datasize.TB.GBytes()` is 1TB represented as GB.

Good catch I need to udpate that

Reply:
Actually to make it clear, maybe declare these as package-level constants? just for clarity

Reply:
Done!